examples/power_method.jl

```diff
@@ -0,0 +1,51 @@
+using Zygote
+using SparseArrays, LinearAlgebra
+using BenchmarkTools
+function power_max_eigen(A, x, target; niter=100)
+    for i=1:niter
+        x = A * x
+        x /= norm(x)
+    end
+    return abs(x' * target)
+end
+
+function power_min_eigen(A, x, target, λ; niter=100)
+    A = λ*I - A 
+    for i=1:niter
+        x = A * x
+        x /= norm(x)
+    end
+    return abs(x' * target)
+end
+
+
+A = sprand(100, 100, 0.1)
+x = randn(100)
+target = randn(100)
+λ = 2
+@btime max_ga_z, max_gx_z, max_gt_z = Zygote.gradient(power_max_eigen, A, x, target) # 3.694 ms (3198 allocations: 12.26 MiB)
```

Comment:
```suggestion
@btime max_ga_z, max_gx_z, max_gt_z = Zygote.gradient(power_max_eigen, $A, $x, $target) # 3.694 ms (3198 allocations: 12.26 MiB)
```

using the `$` sign in your benchmark to avoid type instability. It means evalute and then run the benchmark. Here `A`, `x`, `target` are all global variables.